### PR TITLE
feat(BrowserStorage): remember silent call / message per conversation

### LINKED ISSUE #8336: Silent sent and silent call modes are not behaving the same
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

Yesterday I was very surprised that the "Silent send" option directly sends out the message. I would have assumed that similarly to the "Silent call" option it toggles the button and not send my half baked message already. I assume the problem was we didn't have a button icon for it. So I suggest to use simply the outlined version: https://materialdesignicons.com/icon/send-outline

### LINKED ISSUE #10085: Silent call setting could be remembered per-room
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

**Is your feature request related to a problem? Please describe.**
We have both our team calls and design review calls in the "Design team public" channel. For neither we really want to ring all of the 90+ participants in the channel. We like the silent call feature a lot for that but sometimes we forget to use it.

**Describe the solution you'd like**
Since we always use the silent call feature for every call in that room (except if we forget), it seems it would be apt to remember this setting per-room, much like the video and audio settings are as well.

**Describe alternatives you've considered**
Alternative would be to globally remember, but this does not make a lot of sense as different rooms have different contexts.

### LINKED ISSUE #11591: Remember silent message until I switch conversation
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

**Is your feature request related to a problem? Please describe.**

As a Talk user I sometimes message people when they are busy/offline. I don't want to disturb them because the message isn't urgent. So I send a message without notification aka silent message. So far, so good. But I'm stupid and sometimes I send a second/third message, where I forget to open the actions menu and make the message silent. So I'm sending one silent and one "loud" message. Purpose defeated, the recipient will get a notification.

**Describe the solution you'd like**

Remember my decision for a short time, e.g. until I go to another conversation.

**Describe alternatives you've considered**

Can't think of one.

**Additional context**

N/a

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #10085
* Fix #8336
* Fix #11591
* Based on #11718

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
image | image
image | image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

**Comment by Antreesy** [comment]
Needs a new rebase and dropping a first commit 🙈

File: src/components/MediaSettings/MediaSettings.vue [hunk 1]
```diff
@@ -535,6 +531,16 @@ export default {
...
 
+		setSilentCall(value) {
+			if (value) {
+				BrowserStorage.setItem('silentCall_' + this.token, 'true')
+				this.silentCall = true
+			} else {
+				BrowserStorage.removeItem('silentCall_' + this.token)
+				this.silentCall = false
+			}
+		},
```
Comment (DorraJaouad) [inline src/components/MediaSettings/MediaSettings.vue:542]:
```suggestion
		setSilentCall(value) {
		this.silentCall = value
			if (value) {
				BrowserStorage.setItem('silentCall_' + this.token, 'true')			
			} else {
				BrowserStorage.removeItem('silentCall_' + this.token)
			}
		},
```

File: src/components/NewMessage/NewMessage.vue [hunk 2]
```diff
@@ -994,6 +1012,16 @@ export default {
...
+
+		setSilentChat(value) {
+			if (value) {
+				BrowserStorage.setItem('silentChat_' + this.token, 'true')
+				this.silentChat = true
+			} else {
+				BrowserStorage.removeItem('silentChat_' + this.token)
+				this.silentChat = false
+			}
+		},
```
Comment (DorraJaouad) [inline src/components/NewMessage/NewMessage.vue:1024]:
```suggestion
		toggleSilentChat() {
			this.silentChat = !this.silentChat
			if (this.silentChat) {
				BrowserStorage.setItem('silentChat_' + this.token, 'true')
				this.silentChat = true
			} else {
				BrowserStorage.removeItem('silentChat_' + this.token)
				this.silentChat = false
			}
		},
```

**Review by DorraJaouad — APPROVED** [review]
Tested ✅

**Pull request merged**